### PR TITLE
CTCTRADERS-1002: Unloading FE does not display pointers correctly when receiving an IE058

### PR DESCRIPTION
**Pull request by chris-ianson**
Now reads results of control issues

### CODE REVIEW

**Review by PSouthern — COMMENTED** [review]
@chris-ianson Do we need to extend this for any other errors?  We had this <ErrTypER11>15</ErrTypER11>\n<ErrPoiER12>SLI.Seals number</ErrPoiER12> which was also showing the default pointer.  Although that particular issue may not be displayed anymore following https://github.com/hmrc/declare-transit-movement-unloading-frontend/pull/106

**Review by PSouthern — APPROVED** [review]

**Comment by chris-ianson** [comment]
@PSouthern You've got me thinking now. Maybe the default handler should pass the error message through. Originally we tried to code for what we knew about but it's clear there's other errors we were unaware of and we should at a minimum display the error message. You think I should add it to this PR?

**Comment by PSouthern** [comment]
> @PSouthern You've got me thinking now. Maybe the default handler should pass the error message through. Originally we tried to code for what we knew about but it's clear there's other errors we were unaware of and we should at a minimum display the error message. You think I should add it to this PR?

Yeah, I think so.  Should resolve any other potential errors that are thrown too.

File: app/viewModels/UnloadingRemarksRejectionViewModel.scala [hunk 1]
```diff
@@ -33,15 +33,18 @@ case class UnloadingRemarksRejectionViewModel(page: String, json: JsObject)
...
 
   def apply(errors: Seq[FunctionalError], arrivalId: ArrivalId, nctsEnquiriesUrl: String)(
-    implicit messages: Messages): Option[UnloadingRemarksRejectionViewModel] =
-    errors match {
+    implicit messages: Messages): Option[UnloadingRemarksRejectionViewModel] = {
+    val viewModel: Option[UnloadingRemarksRejectionViewModel] = errors match {
       case error if errors.length == 1 =>
         singleErrorPage(arrivalId, error.head, nctsEnquiriesUrl)
-      case errors if errors.length > 1 =>
+      case `errors` if errors.length > 1 =>
```
Comment (PSouthern) [inline app/viewModels/UnloadingRemarksRejectionViewModel.scala:40]: Why's this enclosed within backticks?

**Review by PSouthern — APPROVED** [review]
See question regarding back ticks.  Functionality is fine though 👍

File: app/viewModels/UnloadingRemarksRejectionViewModel.scala (hunk 1 above)
Comment (chris-ianson) [reply app/viewModels/UnloadingRemarksRejectionViewModel.scala:40]: If not intellij displays a warning about shadowing (removes the warning)

**Comment by chris-ianson** [comment]
@PSouthern I've removed GoodsItemResultsOfControl because it wasn't adding anything (anything that can't be pulled out into a specific ErrorPointer can use the DefaultPointer now)

**Review by PSouthern — APPROVED** [review]

**Pull request merged**